File: Chloe Gwon/2022-06-24-Reverse-Bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int answer = 0;
    
        for (int i=0; i<32; i++){
            answer <<= 1;
            if ((n&1) == 1) answer++;
            n >>= 1;
        }

        return answer;
    }
}

File: somijeon/2022-06-24-reverse-bits.js
// Reverse bits of a given 32 bits unsigned integer.

//* Example 1:

// Input: n = 00000010100101000001111010011100
// Output:    964176192 (00111001011110000010100101000000)

const reverseBits = function (n) {
  let result = 0;

  for (let i = 0; i < 32; i++) {
    // find the last bit of n
    const lastBit = n & 1;

    // shift the last bit of n to the left
    const reversedLastBit = lastBit << (31 - i);

    // insert the reversed last bit of n into the result
    result |= reversedLastBit;

    // the last bit of n is already taken care of, so we need to drop it
    n >>>= 1;
  }

  // convert the result to an unsigned 32-bit integer
  return result >>> 0;
};
